src/CliCommand.php

```diff
@@ -17,39 +17,50 @@ class CliCommand extends \WP_CLI_Command {
   /**
    * Updates products sale discounted percentage.
    */
-  public function refreshSalePercentage(array $args, array $options) {
-    $options = wp_parse_args($options, ['all' => '']);
-    if ($options['all']) {
-      $product_ids = wc_get_products([
-        'return' => 'ids',
-        'limit' => -1,
-      ]);
-      static::updateSalePercentage($product_ids);
-      WP_CLI::success('Sale percentage have been sucessfully updated.');
+  public function refreshSalePercentage(array $args, array $assoc_args) {
+    if ($product_ids = getProductsToUpdate($args, $assoc_args)) {
+      static::udpateProductsMeta($product_ids, 'sale_percentage');
+      WP_CLI::success('Delivery time have been sucessfully updated.');
     }
-    elseif (isset($args[0])) {
-      $product_ids = explode(',', trim($args[0], ','));
-      static::updateSalePercentage($product_ids);
+    else {
+      WP_CLI::error('A comma separated list of products IDs is needed. Use option --all to update all products.');
+    }
+  }
+
+  /**
+   * Updates products delivery time.
+   */
+  public function refreshDeliveryTime(array $args, array $assoc_args) {
+    if ($product_ids = getProductsToUpdate($args, $assoc_args)) {
+      static::udpateProductsMeta($product_ids, 'delivery_time');
+      WP_CLI::success('Delivery time have been sucessfully updated.');
     }
     else {
       WP_CLI::error('A comma separated list of products IDs is needed. Use option --all to update all products.');
     }
   }
 
   /**
-   * Forces the update of products sale discounted percentage.
+   * Updates a given meta field for a list of products.
    *
    * @param array $product_ids
-   *   List of product unique identifiers.
+   *   List of IDs of products to update.
+   * @param string $meta
+   *   Meta field to update.
    */
-  public static function updateSalePercentage($product_ids) {
+  public static function udpateProductsMeta($product_ids, $meta) {
     foreach ($product_ids as $product_id) {
       if (!$product_id = is_numeric($product_id) && (int) $product_id == $product_id ? (int) $product_id : 0) {
```

Comment:
```suggestion
       if (!$product_id = is_numeric($product_id) && (int) $product_id === $product_id ? (int) $product_id : 0) {
```
I don't really understand what this condition is supposed to test though:
```
$ php -r 'var_dump($a = "123", $a == $a, (int) $a == $a, $a == (int) $a, (int) $a === $a, $a === (int) $a);'
string(3) "123"
bool(true)
bool(true)
bool(true)
bool(false)
bool(false)

$ php -r 'var_dump($a = "ab", $a == $a, (int) $a == $a, $a == (int) $a, (int) $a === $a, $a === (int) $a);'
string(2) "ab"
bool(true)
bool(true)
bool(true)
bool(false)
bool(false)
```